src/spirc.rs

```diff
@@ -170,6 +172,16 @@ impl<D: SpircDelegate> SpircManager<D> {
             protocol::spirc::MessageType::kMessageTypePause => {
                 self.delegate.pause();
             }
+            protocol::spirc::MessageType::kMessageTypeNext => {
+            	self.index = (self.index + 1) % self.tracks.len() as u32;
+                let track = self.tracks[self.index as usize];
```

Comment:
Yes, the first track again. The first track in this case is the first track in your playlist/album. Note: if there is only one track in the playlist it is practically in repeat-single mode.